test/src/test/java/hudson/CustomPluginManagerTest.java

```diff
@@ -0,0 +1,157 @@
+            for( ; d!=null; d=d.getParentFile()) {
+                if(new File(d,".jenkins").exists()) {
+                    File dir = new File(d,"war/target/jenkins");
+                    if(dir.exists()) {
+                        System.out.println("Using jenkins.war resources from "+dir);
+                        return dir;
+                    }
+                }
+            }
+            // Temporal solution until JENKINS-34701 is available.
+            throw new AssumptionViolatedException("No jenkins.war resources available");
+        }
+
+        protected ServletContext createWebServer() throws Exception {
+            final File exploded = exploded();
+
+            server = new Server(new ThreadPoolImpl(new ThreadPoolExecutor(10, 10, 10L, TimeUnit.SECONDS, new LinkedBlockingQueue<Runnable>(),new ThreadFactory() {
+                public Thread newThread(Runnable r) {
+                    Thread t = new Thread(r);
+                    t.setName("Jetty Thread Pool");
+                    return t;
+                }
+            })));
+
+            WebAppContext context = new WebAppContext(exploded.getPath(), contextPath);
+            context.setClassLoader(getClass().getClassLoader());
+            context.setConfigurations(new Configuration[]{new WebXmlConfiguration()});
+            context.addBean(new NoListenerConfiguration(context));
+            server.setHandler(context);
+            context.setMimeTypes(MIME_TYPES);
+            context.getSecurityHandler().setLoginService(configureUserRealm());
+            context.setResourceBase(exploded.getPath());
+
+            context.setInitParameter(PluginManager.class.getName() + ".className", CustomPluginManager.class.getClass().getName());
+
+            ServerConnector connector = new ServerConnector(server);
+            HttpConfiguration config = connector.getConnectionFactory(HttpConnectionFactory.class).getHttpConfiguration();
+            // use a bigger buffer as Stapler traces can get pretty large on deeply nested URL
+            config.setRequestHeaderSize(12 * 1024);
+            connector.setHost("localhost");
+            if (System.getProperty("port")!=null)
+                connector.setPort(Integer.parseInt(System.getProperty("port")));
+
+            server.addConnector(connector);
+            server.start();
+
+            localPort = connector.getLocalPort();
+            Logger.getLogger(HudsonTestCase.class.getName()).log(Level.INFO, "Running on {0}", getURL());
+
+            return context.getServletContext();
+        }
+
+    };
+
+    /**
+     * Kills off {@link ServletContextListener}s loaded from web.xml.
+     *
+     * <p>
+     * This is so that the harness can create the {@link jenkins.model.Jenkins} object.
+     * with the home directory of our choice.
+     *
+     * @author Kohsuke Kawaguchi
+     */
+    private static final class NoListenerConfiguration extends AbstractLifeCycle {
```

Comment:
Also to be removed with https://github.com/jenkinsci/jenkins-test-harness/pull/19